src/Morfologik.Stemming/Dictionary.cs

```diff
@@ -80,14 +87,18 @@ public static Dictionary Read(Uri dictURL)
                 throw new IOException("Couldn't construct relative feature map URL for: " + dictURL, e);
             }
 
-            var fsaRequest = (HttpWebRequest)WebRequest.Create(dictURL);
-            var expectedMetadataRequest = (HttpWebRequest)WebRequest.Create(expectedMetadataURL);
+            try
+            {
+                using var httpClient = new HttpClient();
+                using var fsaStream = httpClient.GetStreamAsync(dictURL).GetAwaiter().GetResult();
```

Comment:
Would it be better to make this method async instead? While `GetStreamAsync` uses `.ConfigureAwait(false)` and thus should not risk deadlocking, it might be desirable for callers to be able to await this method. We should make sure to use `.ConfigureAwait(false)` on the calls if this is made async.

Reply:
Would be a breaking change to do so, so could/should be a separate PR/issue.

Reply:
Good idea. Looks like the protocol for maintaining existing synchronous methods for backward compatibility is to add a new method named `ReadAsync()`

```c#
/// <summary>
/// Attempts to load a dictionary using the URL to the FSA file and the
/// expected metadata extension asynchronously.
/// </summary>
/// <param name="dictURL">The URL pointing to the dictionary file (<c>*.dict</c>).</param>
/// <returns>A task representing the asynchronous operation. The result is an instantiated dictionary.</returns>
/// <exception cref="IOException">If an I/O error occurs.</exception>
public static async Task<Dictionary> ReadAsync(Uri dictURL)
{
    Uri expectedMetadataURL;
    try
    {
        string external = dictURL.AbsoluteUri;
        expectedMetadataURL = new Uri(DictionaryMetadata.GetExpectedMetadataFileName(external));
    }
    catch (UriFormatException e)
    {
        throw new IOException("Couldn't construct relative feature map URL for: " + dictURL, e);
    }

    try
    {
        using var httpClient = new HttpClient();
        using var fsaStream = await httpClient.GetStreamAsync(dictURL).ConfigureAwait(false);
        using var metadataStream = await httpClient.GetStreamAsync(expectedMetadataURL).ConfigureAwait(false);

        return Read(fsaStream, metadataStream);
    }
    catch (HttpRequestException e)
    {
        throw new IOException("Error retrieving data from the specified URLs.", e);
    }
}
```

And then call it from the synchronous method

```c#
public static Dictionary Read(Uri dictURL)
{
    return ReadAsync(dictURL).GetAwaiter().GetResult();
}
```

Thoughts?

Reply:
Oh, IAsyncDisposable might not be supported by all targets. If not, then nevermind on that one.

Reply:
I opened #6 to deal with this.